### PR TITLE
RD-3389 Fix hidden names of widgets

### PR DESCRIPTION
**Pull request by leon-good-life**
## Description
In the first commit in this PR, I moved styles from local styles.scss to the AddWidgetsModal But it caused a bug: auto scroll to top. Then I reverted some of the refactoring.
I'm using M1 Mac with Dell monitor 25-inch (2560 × 1440).
To reproduce the bug, I opened Chrome browser, and made 200% Zoom clicking `Cmd` + `+` keys on the keyboard several times.
Before the change there were partially hidden text on headers and buttons were displaced (as seen in the screenshot).
After my changes the text display normally and the location of the buttons are in the modal as expected (as seen in the screenshot).

### Latest PR Fix
https://user-images.githubusercontent.com/17104212/140170263-314bfc4a-b508-4a1e-b7dc-07c1f3a377da.mov

### Previous
Screen Shot 2021-10-26 at 11 09 35
Screen Shot 2021-10-26 at 11 07 50

The following test made on 27-inch (1920 × 1080) Asus monitor with M1 Mac:
https://user-images.githubusercontent.com/17104212/139812074-5c8c1076-fe99-444f-b540-e801d3f5f34a.mov

## Checklist

- [x] My code follows the UI Code Style.
- [x] I verified that all tests and checks have passed.
- [x] I verified if that change requires a documentation update.
- [x] I added proper labels to this PR.

## Tests
System tests:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/1479/pipeline

I'm using M1 Mac with Dell monitor 25-inch (2560 × 1440).
I opened Chrome browser, and made 200% Zoom clicking `Cmd` + `+` keys on the keyboard several times.
Before the change there were partially hidden text on headers and buttons were displaced (as seen in the screenshot).
After my changes the text display normally and the location of the buttons are in the modal as expected (as seen in the screenshot).

 

## Documentation
N/A

### CODE REVIEW

**Comment by leon-good-life** [comment]
I fixed and I tested again:

https://user-images.githubusercontent.com/17104212/139811788-09c439e6-13ca-4200-b6c2-99a261b1fb73.mov


I used 27-inch (1920 × 1080) Asus display with M1 Mac.

File: app/components/AddWidgetModal.tsx [hunk 1]
```diff
@@ -32,6 +34,66 @@ import {
...
+const WidgetList = styled(Item.Group)`
+    height: 435px;
+    overflow: auto;
+    padding-top: 5px;
+
+    @media screen and (max-width: 1281px) {
+        & {
+            margin: 0 !important;
+        }
+    }
```
Comment (qooban) [inline app/components/AddWidgetModal.tsx:55]: Why do we need this?

File: app/components/AddWidgetModal.tsx [hunk 2]
```diff
@@ -32,6 +34,66 @@ import {
...
+    &:hover {
+        background-color: ${colors.greyLight};
+        border-radius: 5px;
+    }
+    .image {
+        & > img {
+            object-fit: contain;
+            height: 80px !important;
+            border-radius: 5px;
+            background: $white;
```
Comment (qooban) [inline app/components/AddWidgetModal.tsx:83]: Let's use:
```suggestion
            background: ${colors.white};
```

File: app/components/AddWidgetModal.tsx [hunk 3]
```diff
@@ -279,148 +341,142 @@ function AddWidgetModal({
...
+        <AddWidgetModalWrapper>
+            <StyledAddWidgetModal
                 trigger={addWidgetBtn}
-                className="addWidgetModal"
                 open={open}
                 closeIcon
                 onOpen={openModal}
                 onClose={closeModal}
                 size="large"
+                className="addWidgetModal"
```
Comment (qooban) [inline app/components/AddWidgetModal.tsx:352]: Do we need it? (as `addWidgetModal` class was removed from `style.scss`)

**Review by qooban — COMMENTED** [review]
Looks a bit better, added some more comments.

My question from previous review is still valid:
> I have also tested the solution in my local environment using RD-3389 branch and found that the second problem described in RD-3389 is still there. Are you planning to fix it within separate PR?

**Comment by leon-good-life** [comment]
@qooban I think I finally fixed it now

**Comment by leon-good-life** [comment]
@qooban Latest PR Fix:
https://user-images.githubusercontent.com/17104212/140170263-314bfc4a-b508-4a1e-b7dc-07c1f3a377da.mov

**Comment by qooban** [comment]
> @qooban Video of the latest PR Fix: https://user-images.githubusercontent.com/17104212/140170263-314bfc4a-b508-4a1e-b7dc-07c1f3a377da.mov

Thanks for letting me know. Is the PR ready for review? If so just use "Re-request review" button, that is clear signal for me that I should jump in 🙂

**Review by qooban — APPROVED** [review]
Looks good from my perspective 👍🏻 

Please run system tests before merging this PR. In case of any failures let me know.

**Comment by leon-good-life** [comment]
@qooban 🙏Thanks a lot for the reviews.
In the future PRs I will make an effort to make the PRs and the commits better in the first place, so they will require less review rounds and will have a clean commit history.

In this PR, I made a mistake when I committed my code, so the commit history doesn’t seem clean to me and I’m not proud of it.

I wonder if I can combine my commits using ‘git rebase --interactive’ (I know it is terrible idea to rewrite history of pushed code) 
Or maybe to archive this PR, delete the branch, and create a new branch and a PR with only one commit. 

Or leave it as is, and just pay more attention to this in the future.

Thanks again.

**Comment by qooban** [comment]
> Thanks a lot for all the reviews. In the future PRs I will make an effort to make the PRs and the commits better in the first place, so they will require less review rounds and will have a clean commit history.

No problem 🙂 You are in introduction phase and that's perfectly understandable.

> In this PR, I made a mistake when I committed my code, so the commit history doesn’t seem clean to me and I’m not proud of it.
> 
> I wonder if I can combine my commits using ‘git rebase --interactive’ (I know it is terrible idea to rewrite history of pushed code) Or maybe to archive this PR, delete the branch, and create a new branch and a PR with only one commit.
> 
> Or leave it as is, and just pay more attention to this in the future.

Don't worry, leave it as is. From my perspective commits history in PR is not as important as commits history on `master` branch. That's why we use "Squash and merge" instead of "Rebase and merge" strategy  when merging PRs to `master` branch (ref. [UI Task Development - Review phase](https://cloudifysource.atlassian.net/wiki/spaces/RD/pages/2124546101/UI+Task+Development#Review-phase) - p. 3). 

And thank you for your comment, it proves that you value quality, and that is very much appreciated 👍🏻

**Comment by qooban** [comment]
@leon-good-life Is there anything left to be done in this PR (as it was already approved)? Are we waiting for something before squash merging?

**Comment by leon-good-life** [comment]
@qooban 
I ran the system tests: [https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/1475/pipeline/55/](https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/1475/pipeline/55/)
Only one test was failed: pages_edit_spec
I run it on my machine on this branch and the test passes

**Comment by qooban** [comment]
@leon-good-life Seems for me that https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/1475/pipeline/55/ was executed without some changes from `master` branch. Please update your branch with latest changes from `master` and re-run system tests. Let's see if that was the problem.

If all fine, then please remember to use **"Squash and merge"** instead of other merging methods.

**Comment by leon-good-life** [comment]
@qooban Thank you. I ran the tests again and they passed!
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/1479/pipeline

**Pull request merged**